vividus-plugin-azure-storage-account/src/main/java/org/vividus/azure/storage/blob/BlobStorageSteps.java

```diff
@@ -231,6 +236,100 @@ public void findBlobs(StringComparisonRule rule, String blobNameToMatch, String
+     *                          </code>
+     *                          <br>
+     *                          <br>where all filters are optional
+     *                          <ul>
+     *                           <li><code>blobNameFilterRule</code> The blob name comparison rule: "is equal to",
+     *                           "contains", "does not contain" or "matches".
+     *                           Should be specified along with <i>blobNameFilterValue</i>.</li>
+     *                           <li><code>blobNameFilterValue</code> The full or partial blob name to be matched.
+     *                           Should be specified along with <i>blobNameFilterRule</i>.</li>
+     *                           <li><code>blobNamePrefix</code> The directory(prefix) to list blobs underneath.</li>
+     *                           <li><code>resultsLimit</code> Maximum number of blob names to return.</li>
+     *                          </ul>
+     * @param containerName     The name of the container to point to.
+     * @param storageAccountKey The key to Storage Account endpoint.
+     * @param scopes            The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of the variable
+     *                          scopes.<br>
+     *                          <i>Available scopes:</i>
+     *                          <ul>
+     *                          <li><b>STEP</b> - the variable will be available only within the step,
+     *                          <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                          <li><b>STORY</b> - the variable will be available within the whole story,
+     *                          <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                          </ul>
+     * @param variableName      The variable name to store the list of found blob names.
+     */
+    @When("I filter blobs by:$filters in container `$containerName` of storage account `$storageAccountKey`"
+            + " and save result to $scopes variable `$variableName`")
+    public void findBlobsByFilters(List<BlobFilter> filters, String containerName,
+                                      String storageAccountKey, Set<VariableScope> scopes, String variableName)
+    {
+        List<BlobFilter> checkedFilters = checkFilters(filters);
+        List<String> blobNames = new ArrayList<>();
+        BlobContainerClient blobContainerClient = createBlobContainerClient(containerName, storageAccountKey);
+        for (BlobFilter filter : checkedFilters)
+        {
+            blobContainerClient
+                    .listBlobsByHierarchy(filter.getBlobNamePrefix().orElse(StringUtils.EMPTY))
+                    .stream()
+                    .map(BlobItem::getName)
+                    .filter(filterMatched(filter))
+                    .limit(filter.getResultsLimit().orElse(Long.MAX_VALUE))
```

Comment:
this approach won't work if only limit is specified: we should rely on server-side limit

Reply:
Server-side limit is `maxResultsPerPage` - but, as discussed with Anton, they don't need limit `per page`, but global one.

Reply:
because the goal of the limit is to reduce the number of results: if the limit is applied on the client side, the client will download all available results at first, and after then limit will be applied - that's performance bottleneck you are trying to eliminate

Reply:
I've re-checked on target huge container - no performance problems with current limit.